Switch over to using cookiecutter

gfontenot opened the pull request:
Using [cookiecutter](https://cookiecutter.readthedocs.io/en/latest/index.html) for our template generation means we don't need to
maintain _any_ code of our own! Huzzah! And we lose zero functionality
in our own template. We actually gain some nice stuff like a template
specific README and LICENSE

gfontenot committed:
Use handlebars (mustache?) for template variables

gfontenot committed:
Nest project in sub-folder

This is the folder that cookiecutter uses to generate the actual
project.

gfontenot committed:
Add cookiecutter config

gfontenot committed:
Add post-generation hooks

This will run bin/setup and initialize an empty git repo after setting
up the project.

gfontenot committed:
Add license and readme for the template

Now that we're using cookiecutter to download/set up the template, it
means we can provide a legit README for the template. This is much
nicer.

gfontenot committed:
Use correct template variables for project name

tonyd256 commented on cookiecutter.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "project_name": "Project Name",
+  "company_name": "thoughtbot",
+  "bundle_identifier": "com.{{ cookiecutter.company_name | replace(' ', '-') | lower }}"
```
should we also be doing this for project name sometimes? I think we need a bundle name as well since spaces won't be allowed in many of the places we have `cookiecutter.project_name`

gfontenot replied:
I was thinking about that. Maybe just removing spaces? I think we can do that in a separate PR if we determine to go down this road further.

georgebrock commented on README.md:
```diff
@@ -1,19 +1,56 @@
 
-This will:
+[suspenders]: https://github.com/thoughtbot/suspenders
+[cookiecutter-django]: https://github.com/thoughtbot/django-template
```
Should `cookiecutter-django` be `django-template` here?

gfontenot replied:
Yep, missed this when I copied it from django-template.

sharplet committed:
Don't include Cartfile.resolved in the template

When installing from the template, the latest matching versions of our
dependencies should be installed.

sharplet committed:
Use Nimble master instead of deleted swift-3.0

sharplet commented:
Tested this out and overall I'm 👍.

I did have some build errors when bootstrapping dependencies (see https://github.com/thoughtbot/kickoff-ios/pull/7), which made me wonder: is there a nice why to re-run the cookiecutter hooks if you get an error? If you have a problem will setting up a repo, it would be great to be able to resume where you left off (or maybe have idempotent setup actions so that they can be re-run safely).

gfontenot commented:
I think we can do whatever we need to do inside the `post_project_gen` hook. I'd say: lets move this forward and then worry about improvements.
